src/Pages/GlobalConfigurations/DeploymentCharts/utils.ts

```diff
@@ -15,33 +15,26 @@
  */
 
 import { SortingOrder } from '@devtron-labs/devtron-fe-common-lib'
-import { sortObjectArrayAlphabetically, versionComparator } from '../common'
-import { ChartDetailType } from './types'
+import { sortObjectArrayAlphabetically, versionComparator } from '@Components/common'
+import { DeploymentChartDTO, DeploymentChartType } from './types'
 
-export const processChartData = (data: ChartDetailType[]): ChartDetailType[] => {
-    let resultData = []
-    const uniqueChartList = new Map<string, ChartDetailType>()
+export const processChartData = (data: DeploymentChartDTO[]): DeploymentChartType[] => {
+    const chartMap: Record<string, DeploymentChartType> = {}
     data.forEach((element) => {
-        const chartDetail = uniqueChartList.get(element.name)
-        if (chartDetail) {
-            chartDetail.count++
-            chartDetail.versions.push({ id: element.id, version: element.version })
-            if (chartDetail.version < element.version) {
-                chartDetail.version = element.version
-                chartDetail.chartDescription = element.chartDescription
-            }
+        const detail = chartMap[element.name]
+        if (detail) {
+            detail.versions.push({ id: element.id, version: element.version, description: element.chartDescription })
```

Comment:
Are we sure we don't need null checks on element, if not then can we de-structure element on params itself?, and also add default value of '' to description